Core/src/main/java/org/tribuo/impl/ArrayExample.java

```diff
@@ -408,14 +415,33 @@ public void set(Feature feature) {
 
     @Override
     public void transform(TransformerMap transformerMap) {
-        for (Map.Entry<String,List<Transformer>> e : transformerMap.entrySet()) {
-            int index = Arrays.binarySearch(featureNames,0,size,e.getKey());
-            if (index >= 0) {
-                double value = featureValues[index];
-                for (Transformer t : e.getValue()) {
+        if(transformerMap.size() < featureNames.length) {
+            //
+            // We have transformers than feature names, so let's
+            // iterate through the map and find the features.
+            for(Map.Entry<String, List<Transformer>> e : transformerMap.entrySet()) {
+                int index = Arrays.binarySearch(featureNames, 0, size, e.getKey());
+                if(index >= 0) {
+                    double value = featureValues[index];
+                    for(Transformer t : e.getValue()) {
+                        value = t.transform(value);
+                    }
+                    featureValues[index] = value;
+                }
+            }
+        } else {
+            //
+            // We have more transformers, so let's fetch them by name.
+            for(int i = 0; i < size; i++) {
+                List<Transformer> l = transformerMap.get(featureNames[i]);
+                if(l == null) {
+                    continue;
```

Comment:
I think I'd prefer this test to be `if (l != null)` and that way we can get rid of the continue statement.